crates/proto/src/udp/udp_stream.rs

```diff
@@ -285,6 +285,24 @@ impl<P: RuntimeProvider> Future for NextRandomUdpSocket<P> {
                         return Poll::Ready(Ok(socket));
                     }
                     Poll::Ready(Err(err)) => match err.kind() {
+                        // See https://github.com/hickory-dns/hickory-dns/issues/2649.
+                        // Windows returns a PermissionDenied error (code 10013) if we attempt
+                        // to bind a port within the ephemeral port range (49,152 - 65535, by default)
```

Comment:
I did some more experimentation locally, and the issue isn't ephemeral ports per se, as I can bind to those successfully.

I tried sweeping through the whole dynamic port range (binding on 127.0.0.1) and got four 10013 permission errors and two 10048 socket in use errors. For all the port numbers that resulted in errors, I checked them with netstat and tasklist, and found that a handful of svchost.exe processes had UDP sockets open for those ports. The ports that produced a permission error each had already-bound sockets on 0.0.0.0 and [::]. The ports that produced "Only one usage of each socket address (protocol/network address/port) is normally permitted." error messages each had already-bound sockets on 127.0.0.1.

I tried replicating binding to 0.0.0.0 with some port number, then binding to 127.0.0.1 with the same port number, but that didn't produce any errors for me. I suspect that this permission error only happens when a more-privileged process has already bound a socket to that port with the unspecified IPv4 or IPv6 address.